src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/PaymentsTest.java

```diff
@@ -0,0 +1,148 @@
+
+    private String getIdamToken() throws com.fasterxml.jackson.core.JsonProcessingException {
+        String idamApiUrl = conf.getString("idam-api-url");
+        String idamClientSecret = conf.getString("idam-client-secret");
+        String username = conf.getString("idam-users-bulkscan-username");
+        String password = conf.getString("idam-users-bulkscan-password");
+        Response idamResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(idamApiUrl)
+            .header(CONTENT_TYPE, APPLICATION_FORM_URLENCODED.getMimeType())
+            .formParam("grant_type", "password")
+            .formParam("redirect_uri", REDIRECT_URI)
+            .formParam("client_id", "bsp")
+            .formParam("client_secret", idamClientSecret)
+            .formParam("scope", "openid profile roles")
+            .formParam("username", username)
+            .formParam("password", password)
+            .post("/o/token");
+
+        assertThat(idamResponse.getStatusCode()).isEqualTo(SC_OK);
+
+        Map<?, ?> r = objectMapper.readValue(idamResponse.getBody().print(), Map.class);
+        return (String)r.get("access_token");
+    }
+
+    private String getS2SToken() throws IOException {
+        String s2sUrl = conf.getString("s2s-url");
+        String s2sSecret = conf.getString("s2s-secret");
+        final String oneTimePassword = format("%06d", new GoogleAuthenticator().getTotpPassword(s2sSecret));
+        Map<String, String> signInDetails = new HashMap<>();
+        signInDetails.put("microservice", "bulk_scan_orchestrator");
+        signInDetails.put("oneTimePassword", oneTimePassword);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        objectMapper.writeValue(baos, signInDetails);
+        String signInDetailsStr = baos.toString();
+
+        Response s2sResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(s2sUrl)
+            .header(CONTENT_TYPE, APPLICATION_JSON.getMimeType())
+            .body(signInDetailsStr)
+            .post("/lease");
+
+        assertThat(s2sResponse.getStatusCode()).isEqualTo(SC_OK);
+
+        return s2sResponse.getBody().print();
+    }
+
+    private Map<?, ?> getCaseData(
+        String accessToken,
+        String s2sToken,
+        String ccdId
+    ) throws JsonProcessingException {
+        String coreCaseDataApiUrl = conf.getString("core-case-data-api-url");
+        Response caseResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(coreCaseDataApiUrl)
+            .header("experimental", true)
+            .header("Authorization", BEARER + accessToken)
+            .header("ServiceAuthorization", BEARER + s2sToken)
+            .get("/cases/" + ccdId);
+
+        assertThat(caseResponse.getStatusCode()).isEqualTo(SC_OK);
+
+        Map<?, ?> c = objectMapper.readValue(caseResponse.getBody().print(), Map.class);
```

Comment:
is it not map with key type String and value type Object ?

Reply:
It gives warning if Map<String,Object>